content/en/language_clients/python/python_client_overview.md

```diff
@@ -0,0 +1,61 @@
+---
+type: docs
+category: Python
+menuTitle: Python Client Overview
+title: Python Client Overview
+weight: 1
+---
+
+[`sigstore`](https://pypi.org/project/sigstore/) is a Python tool for generating and verifying Sigstore signatures. You can use it to sign and verify Python package distributions, or anything else!
+
+## Features
+
+* Supports keyless signature generation and verification
+* Supports signing with ["ambient" OpenID Connect identities](../signing#signing-with-ambient-credentials)
+* A comprehensive [CLI](#usage) and corresponding [importable Python API](https://sigstore.github.io/sigstore-python)
+* An official [GitHub Action](../github_action)
+
+## Usage
+
+For Python API usage, see our [API documentation](https://sigstore.github.io/sigstore-python/).
+
+You can run `sigstore` as a standalone program, or via `python -m`:
+
+```console
+sigstore --help
+python -m sigstore --help
+```
+
+Top-level:
+
+<!-- @begin-sigstore-help@ -->
+```console
+usage: sigstore [-h] [-v] [-V] [--staging | --trust-config FILE] COMMAND ...
+
+a tool for signing and verifying Python package distributions
+
+positional arguments:
+  COMMAND              the operation to perform
+    sign               sign one or more inputs
+    verify             verify one or more inputs
+    get-identity-token
+                       retrieve and return a Sigstore-compatible OpenID
+                       Connect token
+    plumbing           developer-only plumbing operations
+
+optional arguments:
+  -h, --help           show this help message and exit
+  -v, --verbose        run with additional debug logging; supply multiple
+                       times to increase verbosity (default: 0)
+  -V, --version        show program's version number and exit
+  --staging            Use sigstore's staging instances, instead of the
+                       default production instances (default: False)
+  --trust-config FILE  The client trust configuration to use (default: None)
+```
+<!-- @end-sigstore-help@ -->
+
+### SLSA Provenance
+
+This project emits a SLSA provenance on its release! This enables you to verify the integrity of the downloaded artifacts and ensured that the binary's code really comes from this source code.
+
+To do so, please follow the instructions [here](https://github.com/slsa-framework/slsa-github-generator#verification-of-provenance).
```

Comment:
I think this link is out of date (it's wrong in the sigstore-python readme too) - I think it's supposed to be https://github.com/slsa-framework/slsa-github-generator#verify-provenance

It's not very clear from the README, but I think this section about verifying provenance is more closely aligned with the installation section, since it's about verifying the sigstore-python release artifact, not about using it. But they also don't really explain how to find the artifact and attestation. I would almost suggest omitting this section entirely from the docs site because it's not really relevant unless you came across the repo in github and were exploring its release artifacts.

Reply:
My 0.02c is that this can be dropped entirely: the SLSA provenance note is a bit of a red herring in this context, since sigstore-python also emits bundles of its own creation as part of the release process.

(I would like to remove that SLSA provenance step from the release workflow entirely in the near future, since it doesn't currently generate valid SLSA documents - per https://github.com/slsa-framework/slsa-github-generator/issues/3876 - and has been a source of release process breakdowns in the past. I'll likely replace it with GitHub's own attestation actions.)

Reply:
Agreed that this should be in installation.

I would add more details on how to verify the binary, and like Colleen said, where the attestation is (i.e. attached to the release). I'd provide a command using `slsa-verifier` demonstrating how to verify the provenance. Roughly:

1. Download slsa-verifier
2. Download the sigstore-python binary and attestation from a given [release](https://github.com/sigstore/sigstore-python/releases/)
3. `slsa-verifier verify-artifact <path to binary> --provenance-path <path to attestation.intoto.jsonl> --source-uri github.com/sigstore/sigstore-python --source-tag <release tag>` (double check this works)